src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable, inject } from 'inversify';
+import { NotebookCell, NotebookCellOutput, NotebookDocument } from 'vscode';
+import { IVSCodeNotebook } from '../../../common/application/types';
+import { createDeferred, Deferred } from '../../../common/utils/async';
+
+@injectable()
+export class CellOutputDisplayIdTracker {
+    private displayIdCellOutputMappingPerDocument = new WeakMap<
+        NotebookDocument,
+        Map<string, { output: Deferred<NotebookCellOutput>; cell: NotebookCell }>
+    >();
+    private cellToDisplayIdMapping = new WeakMap<NotebookCell, string>();
+    constructor(@inject(IVSCodeNotebook) notebooks: IVSCodeNotebook) {
```

Comment:
What about when a notebook is closed? Shouldn't the mapping be cleared then?

Reply:
Its a weakmap, hence no need.